### ISSUE TITLE
[Project Addition] : Network Anomaly Detection using BiLSTM

### ISSUE THREAD

**Issue by bristiHalder** [abhisheks008/DL-Simplified#863]
### Deep Learning Simplified Repository (Proposing new issue)
:red_circle: **Project Title** : Network Anomaly Detection using BiLSTM </br>
:red_circle: **Aim**: To detect anomalies in network traffic data by leveraging the sequential learning capabilities of LSTM networks, enhanced by the bidirectional processing of BiLSTM.</br>
It is handy for detecting unusual patterns in network traffic that may indicate security threats or malfunctions.

- Data Preprocessing
- BiLSTM Model Design
- Model Training
- Anomaly Detection
- Visualization

**********************************************************************************
### 📍 **Follow the Guidelines to Contribute to the Project:**
* You need to create a separate folder named the Project Title.
* Inside that folder, there will be four main components.
   * Images - To store the required images.
   * Dataset - To store the dataset or, information/source about the dataset.
   * Model - To store the machine learning model you've created using the dataset.
   * `requirements.txt` - This file will contain the required packages/libraries to run the project in other machines.
* Inside the `Model` folder, the `README.md` file must be filled out properly, with proper visualizations and conclusions.

***********************************************************************
:red_circle::yellow_circle: **Points to Note :**

- The issues will be assigned on a first come first serve basis, 1 Issue == 1 PR.
- "Issue Title" and "PR Title should be the same. Include the issue number along with it.
- Follow the Contributing Guidelines & Code of Conduct before starting Contributing.

***********************************************************************
:white_check_mark: **To be Mentioned while taking the issue:**
- Full name: Bristi Halder
- GitHub Profile Link: https://github.com/bristiHalder
- Email ID: bristi18jiya@gmail.com
- What is your participant role? (Mention the Open Source program): GGSOC

*************************************************************
Happy Contributing 🚀 

All the best. Enjoy your open-source journey ahead. 😎

**Comment by abhisheks008** [comment]
Hi @bristiHalder thanks for creating the issue. Apart from BiLSTM what are the other options you are looking for this particular problem statement. Also can you share the source of the dataset?

**Comment by bristiHalder** [comment]
@abhisheks008 
I'm focusing solely on BiLSTM for this problem statement and am not considering other options at this time.

This is the dataset : [Dataset](https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/Advanced_ML_anomaly_detection_L3/DataSets.zip).

**Comment by abhisheks008** [comment]
> @abhisheks008 I'm focusing solely on BiLSTM for this problem statement and am not considering other options at this time.
> 
> This is the dataset : [Dataset](https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/Advanced_ML_anomaly_detection_L3/DataSets.zip).

Focusing on a single algorithm is okay. But here in this project repository we used deploy/implement at least 3 models on the dataset and compare them based on the accuracy scores to find out the best fitted model.

Implementing a single model will not work here.
